modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_3_2/test/TemplateBrowserSnifferUpgradeProcessTest.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.upgrade.v5_3_2.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.dynamic.data.mapping.model.DDMTemplate;
+import com.liferay.dynamic.data.mapping.service.DDMTemplateService;
+import com.liferay.dynamic.data.mapping.service.test.BaseTemplateUpgradeProcessTestCase;
+import com.liferay.fragment.model.FragmentEntry;
+import com.liferay.fragment.service.FragmentEntryService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Albert Gomes Cabral
+ */
+@RunWith(Arquillian.class)
+public class TemplateBrowserSnifferUpgradeProcessTest
+	extends BaseTemplateUpgradeProcessTestCase {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Test
+	public void testUpgradeProcessTemplateRemoveBrowserSniffer()
+		throws Exception {
+
+		DDMTemplate ddmTemplate = addDDMTemplate(
+			".v5_3_2/ddm-template-browser-sniffer-content.ftl");
+
+		FragmentEntry fragmentEntry = addFragmentEntry(
+			".v5_3_2/fragment-entry-browser-sniffer-content.html");
+
+		runTemplateUpgrade();
+
+		ddmTemplate = _ddmTemplateService.getTemplate(
+			ddmTemplate.getTemplateId());
+
+		Assert.assertEquals(
+			read(".v5_3_2/expected-ddm-template-browser-sniffer-content.ftl"),
+			ddmTemplate.getScript());
+
+		fragmentEntry = _fragmentEntryService.fetchFragmentEntry(
+			fragmentEntry.getFragmentEntryId());
+
+		Assert.assertEquals(
+			read(
+				".v5_3_2/expected-fragment-entry-browser-sniffer-content.html"),
+			fragmentEntry.getHtml());
```

Comment:
@alberttgomes, we already declared the services (`DDMTemplateService` and `FragmentEntryService`) in the class `BaseTemplateUpgradeProcessTestCase`, therefore let's reuse it instead of redeclare them here. You can move the code responsible for getting the entities to the base class. Please see:
```
// These add methods we can change their returns to be void
addDDMTemplate(".v5_3_2/ddm-template-browser-sniffer-content.ftl");

addFragmentEntry(".v5_3_2/fragment-entry-browser-sniffer-content.html");

runTemplateUpgrade();

ddmTemplate = getDDMTemplate();

Assert.assertEquals(
	read(".v5_3_2/expected-ddm-template-browser-sniffer-content.ftl"),
	ddmTemplate.getScript());

fragmentEntry = getFragmentEntry();

Assert.assertEquals(
	read(".v5_3_2/expected-fragment-entry-browser-sniffer-content.html"),
	fragmentEntry.getHtml());
```

Reply:
hey @natocesarrego, I just need added in getDDMTemplate and getFragmentEntry, the params `id` because the getTemplate method instanced of DDMtemplate and your respective on FragmentEntry, need your identifier. 
Stay so:
```
DDMTemplate ddmTemplate = addDDMTemplate(
	".v5_3_2/ddm-template-browser-sniffer-content.ftl");

FragmentEntry fragmentEntry = addFragmentEntry(
	".v5_3_2/fragment-entry-browser-sniffer-content.html");

runTemplateUpgrade();

ddmTemplate = getDDMTemplate(ddmTemplate.getTemplateId());

Assert.assertEquals(
	read(".v5_3_2/expected-ddm-template-browser-sniffer-content.ftl"),
	ddmTemplate.getScript());

fragmentEntry = getFragmentEntry(fragmentEntry.getFragmentEntryId());

Assert.assertEquals(
	read(
		".v5_3_2/expected-fragment-entry-browser-sniffer-content.html"),
	fragmentEntry.getHtml());
```